### ISSUE TITLE
If additional dummy data follows, should it be treated as an error or a success?

### ISSUE THREAD

**Issue by IamSukwonKim** [DMTF/libspdm#2321]
I know this isn't a typical situation.

But, what if, responder receives a `GET_VERSION` like `0x10840000FF`?
If it is valid, does that dummy data involved in `M1/M2` and `L1/L2` or not?
If it is invalid, what error code should be responded?

IMO, Error response with `InvalidRequest` would be good. However, it should be stated in the specification.

**Comment by IamSukwonKim** [comment]
>You mean having a non-zero Reserved field?

No, it is little bit different. `Reserved` fields are all specified, but what I am considering is close to _undefined_ field. If we allow that fields, we should prepare huge buffer for huge request/response. (How to implement is out of scope of this issue, though)

For examples, followings should be modified to allow huge request/response.
https://github.com/DMTF/libspdm/blob/567b1c8ea731fe42650d43ede50a105b772dc7aa/include/library/spdm_common_lib.h#L212-L224
https://github.com/DMTF/libspdm/blob/567b1c8ea731fe42650d43ede50a105b772dc7aa/library/spdm_responder_lib/libspdm_rsp_version.c#L88-L89

I don't think it is a good idea. It will cause of waste of resources.

**Comment by steven-bellock** [comment]
Can you give a concrete example of these `undefined` fields? I still don't understand.

**Comment by IamSukwonKim** [comment]
> Can you give a concrete example of these `undefined` fields? I still don't understand.

Sure, as I gave an example at the begin with, `0x10840000FF` is the example.

`GET_VERSION` is defined only 4 Bytes and a field for `byte offset 4` is `undefined`. It seems that requester is allowed to send additional dummy while the message size is less or equal than `DataTransferSize` like `0x10840000FFFF..FFFF`. Furthermore, this additional data should be the input of `M1/M2` and `L1/L2`, so that the codebase should be modified.

**Comment by jyao1** [comment]
> GET_VERSION is defined only 4 Bytes and a field for byte offset 4 is undefined.

Yes, this should be considered as valid, because of the transport layer padding.

> Furthermore, this additional data should be the input of M1/M2 and L1/L2, so that the codebase should be modified.

NO. The SPDM command parser should **ignore** the transport layer padding or dummy bytes after the valid command.
The padding or dummy bytes will **NOT** be in transcript.

**Comment by steven-bellock** [comment]
Ah I see. @IamSukwonKim check out #1837 and #397. In particular, for security reasons, the receiving endpoint may
1. Quickly determine the size of the SPDM message.
2. Zeroize any bytes that aren't part of the SPDM message.

In theory this may mitigate against ROP style attacks. Having said that, SPDM messages should be stored in non-executable data memory. And like @jyao1 said, the transcript is updated only after the SPDM message has been parsed to determine its size, and whether it's a valid message or not.

**Comment by steven-bellock** [comment]
Also note that certain fields, like nonces, can have anything in them, including machine code that can be leveraged by an attacker. So having non-executable memory is the way to go to really be safe.
